### PR TITLE
[FEC-49] - shipping info model

### PR DESCRIPTION
**Pull request by antoniolodias**
#### Summary

This PR creates shipping info model needed for this [migration PR](https://github.com/commercetools/merchant-center-frontend/pull/17434) which is based on [batch 12](https://commercetools.atlassian.net/browse/FEC-49) of the [chapter epic](https://commercetools.atlassian.net/browse/FEC-26) where we are migrating tests to use public test-data package

### CODE REVIEW

File: models/cart/src/shipping-info/builder.spec.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    expect(shippingInfoMock).toEqual(
+      expect.objectContaining({
+        shippingMethodName: shippingMethodName,
+        price: expect.objectContaining({
+          currencyCode,
+        }),
+        // taxedPrice: expect.objectContaining({
+        //   totalNet: expect.objectContaining({ centAmount, currencyCode }),
+        // }),
```
Comment (jaikumar-tj) [inline models/cart/src/shipping-info/builder.spec.ts:93]: can this be uncommented? or was there some issue?

**Review by jaikumar-tj — COMMENTED** [review]
💯 Don't forget to add changeset

**Review by jaikumar-tj — COMMENTED** [review]
This is helpful!
Screenshot 2024-09-04 at 05 19 19

File: models/cart/src/shipping-info/builder.spec.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        taxRate: expect.objectContaining({ name }),
+        taxCategory: expect.objectContaining({ name }),
+        shippingMethod: expect.objectContaining({ name }),
+        deliveries: expect.arrayContaining([expect.objectContaining({ id })]),
+        // discountedPrice: expect.objectContaining({
```
Comment (CarlosCortizasCT) [inline models/cart/src/shipping-info/builder.spec.ts:98]: Same comment as above.

File: models/cart/src/shipping-info/types.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import type { ShippingInfo } from '@commercetools/platform-sdk';
+import type { TBuilder } from '@commercetools-test-data/core';
+
+export type TShippingInfo = ShippingInfo;
+
+export type TShippingInfoGraphql = TShippingInfo & {
```
Comment (CarlosCortizasCT) [inline models/cart/src/shipping-info/types.ts:6]: The Graphql representation has different value types for several fields and also has some extra ones so we need to define them here.

image

File: models/cart/src/shipping-info/transformers.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+const transformers = {
+  default: Transformer<TShippingInfo, TShippingInfo>('default', {
+    buildFields,
+  }),
+  rest: Transformer<TShippingInfo, TShippingInfo>('rest', {
+    buildFields,
+  }),
+  graphql: Transformer<TShippingInfo, TShippingInfoGraphql>('graphql', {
+    buildFields,
+    addFields: () => ({
```
Comment (CarlosCortizasCT) [inline models/cart/src/shipping-info/transformers.ts:24]: We will need to use the `replaceFields` prop to handle the difference in values for some of the types and also adding the ones which are specific to the GraphQL representation.

As an example, `taxCategory` is of type `TaxCategoryReference` in the REST representation but it is a `TTaxCategoryGraphql` type for the GraphQL. Also, in GraphQL we have the `taxCategoryRef` extra property of type `TReferenceGraphql`.

You can take a look at the [ProductProjection transformers](https://github.com/commercetools/test-data/blob/main/models/product-projection/src/transformers.ts#L95) to get an example of how to manage this.

File: models/cart/src/shipping-info/builder.spec.ts (hunk 1 above)
Comment (CarlosCortizasCT) [reply models/cart/src/shipping-info/builder.spec.ts]: This should be working now after adding missing models and adjusting tests in this PR: 4a33806b5e146a883738cd73aa935d7942359bdd

File: models/cart/src/shipping-info/builder.spec.ts (hunk 2 above)
Comment (CarlosCortizasCT) [reply models/cart/src/shipping-info/builder.spec.ts]: This should be working now after adding missing models in this PR: 4a33806b5e146a883738cd73aa935d7942359bdd

**Comment by CarlosCortizasCT** [comment]
After all the changes we've introduced to the way models should be implemented, we decided to start over this initiative on this PR: #726

**Pull request closed without merging**